state/application_test.go

```diff
@@ -271,7 +271,10 @@ func (s *ApplicationSuite) TestCAASSetCharm(c *gc.C) {
 	})
 	defer st.Close()
 	f := factory.NewFactory(st, s.StatePool)
-	app := f.MakeApplication(c, &factory.ApplicationParams{Name: "mysql"})
+	app := f.MakeApplication(c, &factory.ApplicationParams{Name: "mysql", CharmOrigin: &state.CharmOrigin{
+		Source:   "local",
+		Platform: &state.Platform{OS: "ubuntu", Channel: "20.04"},
+	}})
```

Comment:
This was the only test that failed as a result of the factory change.

The reason for this the alignment of a call to `MakeApplication` to a CAAS model without a specific charm in the ApplicationParams, leading to the cham having series focal, but the application quantal

The test can be restored by including a charm origin in MakeAppliation, rather than falling back to the default